controller/scheduler/swarm.py

```diff
@@ -32,11 +43,15 @@ def create(self, name, image, command='', template=None, **kwargs):
             if mem[-2:-1].isalpha() and mem[-1].isalpha():
                 mem = mem[:-1]
         cpu = kwargs.get('cpu', {}).get(l['c_type'])
-        self.docker_cli.create_container(image=cimage, name=name,
-                                         command=command.encode('utf-8'), mem_limit=mem,
-                                         cpu_shares=cpu,
-                                         environment=[affinity],
-                                         host_config={'PublishAllPorts': True})
+        try:
+            self.docker_cli.create_container(image=cimage, name=name,
+                                             command=command.encode('utf-8'),
+                                             mem_limit=mem,
+                                             cpu_shares=cpu,
+                                             environment=[affinity],
+                                             host_config={'PublishAllPorts': True})
+        except Exception as err:
```

Comment:
We should catch `DockerException` rather than `Exception`--it's almost always bad Python form to catch every exception. (I will make this same comment every time I see a catch-all `except`.)